### PR TITLE
Add local fix for torch.save on XLA tensors

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: test/test_operations.py [hunk 1]
```diff
@@ -570,6 +570,14 @@ def test_fn(a):
 
     self.runAtenTest(torch.rand(4, 3), test_fn)
 
+  def test_save(self):
+    xla_device = xm.xla_device()
+    x = torch.randn(5, device=xla_device)
+    x_file = '/tmp/x.pt'
```
Comment (dlibenzi) [inline test/test_operations.py]: I'd do:
```Python
  temp_file = tempfile.mktemp()
  try:
    ...
  finally:
    os.remove(temp_file)
```

File: torch_patches/X10-torch_save.diff [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+index 43be96d70..0cd8aaae5 100644
+--- a/torch/serialization.py
++++ b/torch/serialization.py
+@@ -221,6 +221,8 @@ def save(obj, f, pickle_module=pickle, pickle_protocol=DEFAULT_PROTOCOL):
+         >>> buffer = io.BytesIO()
+         >>> torch.save(x, buffer)
+     """
++    if torch.is_tensor(obj) and obj.device.type == 'xla':
++        obj = obj.cpu()
+     return _with_file_like(f, "wb", lambda f: _save(obj, f, pickle_module, pickle_protocol))
```
Comment (dlibenzi) [inline torch_patches/X10-torch_save.diff]: What is the story here?
Should we have this as official PR?

Comment (asuhan) [reply torch_patches/X10-torch_save.diff]: I think there might be concerns around serializing tensors which share the storage multiple times. @li-roy probably knows more about this.

File: test/test_operations.py (hunk 1 above)
Comment (asuhan) [reply test/test_operations.py]: Done.

**Comment by dlibenzi** [comment]
Just make sure we get a cleaner path into PT to solve this one.

**Pull request merged**